### DISCUSSION TITLE
Any one pager doc/readme to help new developers trying to learn running MinIO from sources?

### DISCUSSION THREAD

**Discussion by grecinto** [minio/minio#20665] [Q&A] [answered]
Yep, 'wanting to learn running from sources (as a developer doing changes in the UI for example), 'just for my educational/fun dev't purposes. I am sure this was asked numerous times before, pardon for redundancy.

Kind Regards,
-Gerardo

**Comment by klauspost** [comment]
A) Install [Go](https://go.dev/dl/)
B) Run `go build` in the root.
C) `./minio server ...`

There isn't really much more to it.

**Reply by grecinto** [reply]
No, I am looking for tips navigating code so I can make changes in the UI, as I described. I am able to create a debug harness in VSCODE, but yeah, the code seems a little obfuscated, I have not seen this kind of coding standard.

Anyway, I appreciate any help/tips.

Kind Regards,
-Gerardo

**Reply by klauspost** [reply, marked as answer]
In any order:

A) Read the [docs](https://docs.min.io) to get an idea of the server structure.
B) See about the [S3 API](https://docs.aws.amazon.com/AmazonS3/latest/API/API_Operations_Amazon_Simple_Storage_Service.html).
C) Chose a starting point. For example [the router](https://github.com/minio/minio/blob/master/cmd/api-router.go#L298) and follow the code down the call path.

There is no magic to understanding the code. There is just a lot of it.

**Comment by marktheunissen** [comment]
@grecinto If you want to change the UI you might be talking about Console, which is a separate repo: https://github.com/minio/console

It's built into the minio binary as a Go dep, so just wire up your local environment appropriately, using `replace` in go.mod

**Comment by grecinto** [comment]
Thanks a lot to all of you, I appreciate the response. @marktheunissen is right, I am in the wrong GitHub repo. lol.
But nothing big deal, I was just fascinated and tried to learn more about MinIO.

I am not a reactJS/NodeJS expert, thus, ... perhaps in the future if I learn those I'll play around. :)

Kind Regards.
